### ISSUE TITLE
ESP32-C6 support

### ISSUE THREAD

**Issue by probonopd** [espressif/arduino-esp32#7713]
### Related area

ESP32-C6 

### Hardware specification

Support for ESP32-C6

### Is your feature request related to a problem?

ESP32-C6 is not supported yet

### Describe the solution you'd like

Support for ESP32-C6

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

**Comment by probonopd** [comment]
https://www.espressif.com/en/products/socs/esp32-c6

**Comment by lbernstone** [comment]
C6 will not be supported until ESP-IDF 5.1, so arduino will not support it until that is released.

**Comment by VojtechBartoska** [comment]
Hello, support of ESP32-C6 is on our Roadmap, @lbernstone is right that it will be supported in ESP-IDF 5.1 so in Arduino, it will be in version 3.X.X. We'll soon create development branch and we will start working on IDF 5 support there.

@probonopd Thanks for opening the issue, we will use it to track general support ESP32-C6 :)

**Comment by tuct** [comment]
My first esp32 c6 dev boards v0.0 arrived today, eager to test it!

**Comment by simogaspa84** [comment]
Hi @tuct how did you get it the esp32 c6 dev board? which supplier ? thanks

**Comment by simogaspa84** [comment]
Thanks a lot @negis02 .. very kind... are you going to try the new features about cyber security in the C6 dev board? 
I am really interested about the trusted execution enviroment and the glitch detector.. Do you know something about that?
Thanks

**Comment by seajhawk** [comment]
Hey @simogaspa84 - thanks for posting.
 I'm mainly interested in the Thread support, but I'll have to look at the trusted execution and glitch detector features. I wasn't even aware of those!

I'm building an air quality, temp, humidity, CO2 sensor. 

Does anyone know if we can use a C3 board for basic prototyping until the C6 support is added?

**Comment by tuct** [comment]
I got mine from the official store at aliexpress
€ 9,81  1%OFF | ESP32-C6-DevKitC-1 Entwicklung Bord Espressif Systeme ESP32-C6 Serie * Probe * Limit 2 stücke pro Kunden * (siehe beschreibung)
https://a.aliexpress.com/_mLI99g2
I bought 5, so the limit is new, happy to share of you are close to or in Vienna

**Comment by Trevo525** [comment]
You can also purchase from Adafruit's [official store](https://www.adafruit.com/product/5672). They have 47 in stock at the time of writing this.

**Comment by a-d-r-i-a-n-d** [comment]
I've ordered mine today, looking forward to C6 support.

**Comment by MingyaoLiu** [comment]
I just got my esp32-c6 wroom module, looking forward to have arduino capability!

**Comment by guumeyer** [comment]
I just got my esp32-c6 wroom module too, looking forward to have arduino capability!

**Comment by N0ury** [comment]
idf.py --version
ESP-IDF v5.1-dev-3710-gacac972f70

It looks like v5.1 has been released ;-)

**Comment by me-no-dev** [comment]
It has not. This is just the version that will be used in the future. 5.1 is in ESP-IDF master and needs to be split in it's own branch before it will be released. It has not even been feature-frozen yet :)

**Comment by djmjr77** [comment]
Update please?

**Comment by me-no-dev** [comment]
Guys, please stop asking for update every few days. Before we add support for new chips, there is a lot of work that needs to be done. When that is done (will take way more than few days) most if not all new chips will be supported. We will announce that, so even less reason to keep asking.

**Comment by Hedda** [comment]
> Hello, support of ESP32-C6 is on our Roadmap, @lbernstone is right that it will be supported in ESP-IDF 5.1 so in Arduino, it will be in version 3.X.X. We'll soon create development branch and we will start working on IDF 5 support there.
> 
> @probonopd Thanks for opening the issue, we will use it to track progress on support of ESP32-C6 :)

As noted, the official Espressif ESP-IDF v5.1, which is currently in development, will include initial support for the new ESP32-C6 chip, however, while Espressif ESP-IDF v5.1 has not yet been released it should work if using the master branch in esp-idf repository. If you want to track the development status then follow these the updates made to these other issues for ESP32-C6:

https://github.com/espressif/arduino-esp32/issues/7852

https://github.com/espressif/esp-idf/issues/10423

**Comment by LenReinhart** [comment]
Another waiting for C6 support. Subscribed.

**Comment by SuGlider** [comment]
Ok, we have C6 support already. You can try the C6 branch:
https://github.com/espressif/arduino-esp32/tree/feature/esp32c6-support

**Comment by SuGlider** [comment]
This the initial stage of support. At this time we are working in testing and fixing.

UART is fully tested and everything works correclty, including using UART with `setCpuFrequencyMhz()` at 160, 80, 40, 20 and 10MHz.

The same for RMT.

**Comment by SuGlider** [comment]
I have tested the Basic WiFi and BLE examples and those are working fine with the current C6 branch.
At this point, please, don't expect to see low power consumption yet.

We still must work on the IDF Power Management component of the Arduino Core for the ESP32-C6.
It will take some time, but we expect to offer similar IDF results to the Arduino Users related to power consumpion in IoT applications with the C6, and maybe also for other SoCs.

Especificaly for the C6, we will also work on the WiFi 6 (802.11ax), adding an Arduino API for controlling the TWT (Target Wait Time) configuration.


#### Some IDF examples using the above referenced IDF APIs:
IDF Power Management Example:
https://github.com/espressif/esp-idf/blob/master/examples/wifi/power_save/README.md

IDF ESP32-C6 WiFi 6 802.11ax Example:
https://github.com/espressif/esp-idf/tree/master/examples/wifi/itwt

**Comment by LenReinhart** [comment]
How do I integrate this with the Arduino application?

**Comment by lbernstone** [comment]
@LenReinhart This is a very early release.  It is intended for the people that will build those pieces. As suglider says, it will take some time.

**Comment by SuGlider** [comment]
> How do I integrate this with the Arduino application? 

If you are talknig about how to "install" the C6 Arduino Core branch, it is necessary to use it as a `sketchbook board` by cloning this branch into the local Sketchbook Folder. The process of cloning and creating the `sketchbook board` is described in this link:

https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html#how-to-update-to-the-latest-code

**Comment by LenReinhart** [comment]
Understood. I have a copy of Dr. Ting’s Forth that was designed for the ESP32 dev board. It uses Arduino to build the image download. I correctly guessed that the front end would be as much the same as possible on the compiler and it would take care of the translation. I successfully compiled and loaded it on the C3 dev board simply by changing the target. I have two of the C6 boards and would like to see if it will work for that. If it doesn’t no big deal. Bluetooth did not work, but I didn’t expect it would as it was the LE version. I’m 76 and have been working with computer systems since the 8080, now I just like to mess around with stuff and I thought what was done with using the  Arduino system, to build and install a copy of Forth OS, I thought was very clever. The fact that it has the possibility of being very transportable among processors using the same compiler and the way the Arduino system takes care of the addressing differences between hardware is very nice IMO.

**Comment by lbernstone** [comment]
@LenReinhart Sounds good.  Build it on the C3, which has a stable build ATM :)

**Comment by VojtechBartoska** [comment]
First initial support PR: https://github.com/espressif/arduino-esp32/pull/8337

**Comment by VojtechBartoska** [comment]
## Table with Support overview

_We will keep it up to date.

Date of last update: 26.6.2023_

| **ESP32-C6 Support Overview**   |   Assignee   |                |             |              |
|---------------------------------|:------------:|:--------------:|:-----------:|:------------:|
|          **Peripheral**         |              |   **Testing**  | **Updates** | **Comments** |
|               ADC               | @P-R-O-C-H-Y |                |             |              |
|          Ethernet  SPI          |              |                |             |              |
|               GPIO              | @P-R-O-C-H-Y |                |             |              |
|               GDMA              |              |                |             |              |
|       I2C (Slave & Master)      |              |                |             |              |
|               I2S               |              |                |             |              |
|             LED PWM             |              |                |             |              |
|              MCPWM              |              |                |             |              |
|               PCNT              |              |                |             |              |
|        Parallel IO driver       |              |                |             |              |
|               RMT               |              |                |             |              |
|          SDIO 2.0 Slave         |              |                |             |              |
|               SPI               |              |                |             |              |
|               TWAI              |              |                |             |              |
|           Temp. Sensor          |              |                |             |              |
|               UART              |   @SuGlider  |    Fully Ok    |             |              |
|         USB Serial/JTAG         |              |                |             |              |
|            SigmaDelta           | @P-R-O-C-H-Y |                |             |              |
| Timers (System/Generel-purpose) | @P-R-O-C-H-Y |                |             |              |
|              Touch              |              |                |             |              |
|           **General**           |              |                |             |              |
|              Wi-Fi              |   @SuGlider  | Examples works |             |              |
|            Bluetooth            |              |                |             |              |
|           Bluetooth LE          |   @SuGlider  | Examples works |             |              |
|             802.15.4            |              |                |             |              |
|              Thread             |              |                |             |              |
|              Zigbee             |              |                |             |              |

**Comment by lboue** [comment]
> C6 will not be supported until ESP-IDF 5.1, so arduino will not support it until that is released.

[ESP-IDF Release v5.1](https://github.com/espressif/esp-idf/releases/tag/v5.1) has been released.

**Comment by rdahlin** [comment]
> > C6 will not be supported until ESP-IDF 5.1, so arduino will not support it until that is released.
> 
> @lbernstone [ESP-IDF Release v5.1](https://github.com/espressif/esp-idf/releases/tag/v5.1) has been released.

Any news here?

**Comment by lbernstone** [comment]
Pre-release support as shown in the table above is in the [idf-v5.1 branch](https://github.com/espressif/arduino-esp32/tree/esp-idf-v5.1-libs)

**Comment by VojtechBartoska** [comment]
Hello, we are now collecting feedback on release 2.0.11 and when we are good to go, above mentioned branch https://github.com/espressif/arduino-esp32/tree/esp-idf-v5.1-libs will be merged into master and initial support for ESP32-C6 & ESP32-H2 will be available for development purposes.

**Comment by petergjoel** [comment]
@VojtechBartoska in `HardwareSerial.cpp` I find `#define RX1 5` and `#define TX1 4` for UART1 (LP_UART pins) - this is conflicting with the documentation.

However, using the `#define RX1 5` and `#define TX1 4` works fine for our application -- as long as we flip the wiring accordingly.

**Comment by petergjoel** [comment]
See also issue with number of UART ports and the esp32-c6: https://github.com/espressif/esp-idf/issues/11989#issue-1830756977

**Comment by HX2003** [comment]
Hi all, I have successfully flashed the ESP32-C6 using https://github.com/espressif/arduino-esp32/tree/esp-idf-v5.1-libs. Using ble Unified Provisioning I got [`Configuration mismatch. Select BLE Only or BTDM mode from menuconfig`](https://github.com/espressif/esp-idf/blob/cb174b0fe1f150f2ecc2ce87e0f02154405c6677/components/protocomm/src/simple_ble/simple_ble.c#L228C20-L228C88), this error does not occur on the ESP32-S3 using the same build. I wish to continue to use Arduino IDE and not setup platform io.

**Comment by lbernstone** [comment]
> Hi all, I have successfully flashed the ESP32-C6 using [`esp-idf-v5.1-libs`](https://github.com/espressif/arduino-esp32/tree/esp-idf-v5.1-libs?rgh-link-date=2023-09-29T18%3A04%3A50Z). Using ble Unified Provisioning I got [`Configuration mismatch. Select BLE Only or BTDM mode from menuconfig`](https://github.com/espressif/esp-idf/blob/cb174b0fe1f150f2ecc2ce87e0f02154405c6677/components/protocomm/src/simple_ble/simple_ble.c#L228C20-L228C88), this error does not occur on the ESP32-S3 using the same build. I wish to continue to use Arduino IDE and not setup platform io.

Looks like this is a problem upstream.  There are no CTRL_MODE settings in C6 menuconfig.

**Comment by HX2003** [comment]
@lbernstone I am directly invoking from esp idf
`#include <wifi_provisioning/scheme_ble.h>`
`#include <wifi_provisioning/manager.h>`
Although I assume this problem would also occur if one uses `libraries/WiFiProv/examples/WiFiProv` as well.

**Comment by IsraelMek** [comment]
Hi, I have added ESP32 core 2.0.14 to the boards manager. Even though in github ESP32C6 is mentioned on the list of supported boards, it does not show up in the selection list of Arduino.

**Comment by me-no-dev** [comment]
@IsraelMek you need 3.0.0 for that. It is still in beta and you need to use different BoardManger URL to get it: https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json

**Comment by IsraelMek** [comment]
@me-no-dev Thanks a lot . Update done to 3.0.0 and I can see ESP32-C6 in the board selection. But it doesn't compile the builtin examples for when C6 is selected. it displays error message "bits/c++config.h: No such file or directory". For ESP32-S3, the IDE compiles the example without any issue.

**Comment by me-no-dev** [comment]
It seems that Arduino has some problems extracting the toolchains properly... especially riscv one. Will investigate Monday and come back. You could open the json in the link and download and manually install the toolchain till then

**Comment by me-no-dev** [comment]
RV toolchains:
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-aarch64-linux-gnu.tar.xz
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-arm-linux-gnueabi.tar.xz
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-i686-linux-gnu.tar.xz
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-x86_64-apple-darwin.tar.xz
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-aarch64-apple-darwin.tar.xz
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-i686-w64-mingw32.zip
https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/riscv32-esp-elf-12.2.0_20230208-x86_64-w64-mingw32.zip

**Comment by IsraelMek** [comment]
@me-no-dev The manual installation fixed it. Thanks once again.

**Comment by Hook654321** [comment]
How do I have to install it manually?

**Comment by Atomique13** [comment]
> @me-no-dev Thanks a lot . Update done to 3.0.0 and I can see ESP32-C6 in the board selection. But it doesn't compile the builtin examples when C6 is selected. it displays error message "bits/c++config.h: No such file or directory". For ESP32-S3, the IDE compiles the example codes without any issue.

Same problem on 3.0.0 with C3.

**Comment by me-no-dev** [comment]
> Same problem on 3.0.0 with C3 and H2.

All RiscV chips currently use the same exact toolchain, so yes. same on C3, C6 and H2

**Comment by me-no-dev** [comment]
We have found the problem, but not a solution yet. The path to some files inside the toolchain becomes too long for gcc to handle and it does not see the file, though it's there. When installed from git, this issue does not occur, because path is short enough to not trigger it. We do not control where the toolchains are installed, so ... fingers crossed 😆

**Comment by TD-er** [comment]
So installing it to something like `C:\A\` (shortest path possible) could be a work-around?

Or is it a problem where the command line of some commands gets too long? (including all paths needed for the command line)

**Comment by me-no-dev** [comment]
It's the path to the header that get's too long. GCC seems to limit it to maximum 200 symbols (including the drive). This is how the path looks: 
`c:/users/username/appdata/local/arduino15/packages/esp32/tools/riscv32-esp-elf-gcc/esp-12.2.0_20230208/riscv32-esp-elf/include/c++/12.2.0/riscv32-esp-elf/rv32imc_zicsr_zifencei/ilp32/no-rtti/bits/c++allocator.h`

**Comment by TD-er** [comment]
I do see a lot of "riscv32-esp-elf" being used in the path.
Isn't that a bit too verbose (and obvious enough) ?
Also "rv32" is being used, so this suggests it can be shortened without loosing its meaning.

**Comment by TD-er** [comment]
By the way, not sure if it is a limit of GCC. I think it might be a limit of the (Windows) shell environment of 255 bytes.
If I remember correctly there was a work-around for this as there have been several of these limits in the past. The next one being 32k (not sure if 32k Unicode chars or 32k bytes)

**Comment by me-no-dev** [comment]
this is the structure of the toolchain. There are reasons for it and we do not control it, nor can alter it. We control only `riscv32-esp-elf-gcc/esp-12.2.0_20230208` which is the package name and version. We can shorten it slightly, but that would not help for longer usernames... we will see how we will solve this.

**Comment by me-no-dev** [comment]
> By the way, not sure if it is a limit of GCC.

It is. Windows supports longer paths than 200 symbols. You can access the files otherwise. Only GCC does not see them.

**Comment by TD-er** [comment]
I know Windows does support it, but some shell environments do have different limits.
Not sure if this differs between `cmd` and PowerShell , but that would be an easy fix if these actually differ here.

**Comment by omerk** [comment]
As expected, `3.0.0-alpha1` with Arduino IDE 2.2.1 on Windows fails to compile:
```
In file included from c:\users\omerk\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\esp-12.2.0_20230208\riscv32-esp-elf\include\c++\12.2.0\stdlib.h:36,
                 from C:\Users\omerk\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-f0437b945f/esp32c6/include/newlib/platform_include/assert.h:13,
                 from c:\users\omerk\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\esp-12.2.0_20230208\riscv32-esp-elf\sys-include\sys\reent.h:503,
                 from C:\Users\omerk\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-f0437b945f/esp32c6/include/newlib/platform_include/sys/reent.h:9,
                 from c:\users\omerk\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\esp-12.2.0_20230208\riscv32-esp-elf\sys-include\stdio.h:60,
                 from C:\Users\omerk\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-alpha1\cores\esp32/Arduino.h:27,
                 from C:\Users\omerk\AppData\Local\Temp\arduino\sketches\9EA17A2B30510F72480E2CA2A9127035\sketch\Blink.ino.cpp:1:
c:\users\omerk\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\esp-12.2.0_20230208\riscv32-esp-elf\include\c++\12.2.0\cstdlib:41:10: fatal error: bits/c++config.h: No such file or directory
   41 | #include <bits/c++config.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

exit status 1

Compilation error: exit status 1
```

Using `arduino-cli` inside WSL (Ubuntu 20.04) works just fine though:
```
$ arduino-cli compile --fqbn esp32:esp32:esp32c6 --build-path blink-c6/bin blink-c6
Sketch uses 181666 bytes (13%) of program storage space. Maximum is 1310720 bytes.
Global variables use 8928 bytes (2%) of dynamic memory, leaving 318752 bytes for local variables. Maximum is 327680 bytes.

Used platform Version      Path
esp32:esp32   3.0.0-alpha1 /home/omer/.arduino15/packages/esp32/hardware/esp32/3.0.0-alpha1
```

Following [these instructions](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=powershell) didn't fix the issue.

**Comment by IsraelMek** [comment]
@Hook654321 For the manual fix,  I search and copied the missing header file(c++config.h) to the folder indicated in the error message which is c:\users\user_name\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\esp-12.2.0_20230208\riscv32-esp-elf\include\c++\12.2.0\bits.

**Comment by sfe-SparkFro** [comment]
FWIW my workaround is to simply rename the folder `esp-12.2.0_20230208` to `esp-12.2.0`. This shortens the path length just enough for GCC to find the needed files, no need to manually install anything anywhere. I understand this isn't a perfect solution for everyone (eg. those with long usernames), but might be helpful for folks wanting to use it right now.

**Comment by Guytou11** [comment]
Hi,  Is there any update on when Arduino will support the ESP32-C6 ?    Is there perhaps a Zigbee library available that could be used on PlatformIO for the ESP32-C6 ?

**Comment by thhkd** [comment]
> Hi, Is there any update on when Arduino will support the ESP32-C6 ? Is there perhaps a Zigbee library available that could be used on PlatformIO for the ESP32-C6 ?

ESP32-C6 y supported in arduino IDE using the board extension ESP32 by Espressif Systems version 3.0.0 alpha2. I hope it won't take long to have it available in PlatformIO aswel.

**Comment by VojtechBartoska** [comment]
Hi, yes, you can already used development branch in Arduino IDE for ESP32-C6 support, keep in mind that not everything is supported yet.

**Comment by ITstreet1** [comment]
I have an official ESP32-C6 DevKitM and use the latest alpha release available on Arduino IDE boards manager.
https://docs.espressif.com/projects/espressif-esp-dev-kits/en/latest/esp32c6/esp32-c6-devkitm-1/index.html
When connected over the UART port it shows up as a COM port
When connected over the USB port it shows up as an ESP32H2 Dev Module.
Obviously, it is not an H2.

I tried WiFi Scan. Works!
I tried the Adafruit_Neopixel Simple example for testing the addressable LED on pin 8. Don't work!
I get:

> ESP-ROM:esp32c6-20220919
> Build:Sep 19 2022
> rst:0xc (SW_CPU),boot:0xc (SPI_FAST_FLASH_BOOT)
> Saved PC:0x4001975a
> SPIWP:0xee
> mode:DIO, clock div:2
> load:0x4086c410,len:0xc24
> load:0x4086e610,len:0x26f8
> load:0x40875728,len:0x594
> entry 0x4086c410
> E (123) rmt(legacy): CONFLICT! driver_ng is not allowed to be used with the legacy driver
> 
> abort() was called at PC 0x42005dfd on core 0
> Core  0 register dump:
> MEPC    : 0x40802168  RA      : 0x40806142  SP      : 0x4087e100  GP      : 0x4080c0d0  
> TP      : 0x00000000  T0      : 0x37363534  T1      : 0x7271706f  T2      : 0x33323130  
> S0/FP   : 0x4087e12c  S1      : 0x4087e12c  A0      : 0x4087e138  A1      : 0x4087e11a  
> A2      : 0x00000000  A3      : 0x4087e165  A4      : 0x00000001  A5      : 0x4080e000  
> A6      : 0x00000000  A7      : 0x76757473  S2      : 0x42023000  S3      : 0x4080d734  
> S4      : 0x42020000  S5      : 0x42000000  S6      : 0x4080217a  S7      : 0x00019590  
> S8      : 0x0000c4e0  S9      : 0x00000000  S10     : 0x00000000  S11     : 0x00000000  
> T3      : 0x6e6d6c6b  T4      : 0x6a696867  T5      : 0x66656463  T6      : 0x62613938  
> MSTATUS : 0x00001881  MTVEC   : 0x40800001  MCAUSE  : 0x00000007  MTVAL   : 0x00000000  
> MHARTID : 0x00000000  
> 
> Stack memory:
> 4087e100: 0x00000000 0x00000000 0x4087e12c 0x4080a8d4 0x4080217a 0x42000000 0x42020030 0x4080c658
> 4087e120: 0x4087e12c 0x4080c674 0x4087e118 0x30303234 0x64666435 0x28010200 0x726f6261 0x20292874
> 4087e140: 0x20736177 0x6c6c6163 0x61206465 0x43502074 0x34783020 0x35303032 0x20646664 0x63206e6f
> 4087e160: 0x2065726f 0x00000030 0x42020000 0xabe96d90 0x20001090 0x4202a394 0x4202a398 0x42005e00
> 4087e180: 0x42023000 0x4202a394 0x4202a3a8 0x4200875e 0x00000006 0x00000000 0x00000000 0x4200a944
> 4087e1a0: 0x00000000 0x00000000 0x00700000 0x420093de 0x0000004d 0x0000000c 0x0000004d 0x42000000
> 4087e1c0: 0x0000004d 0x0000000c 0x0000004d 0x4080230c 0x00000000 0x00241900 0x00000000 0x02625a00
> 4087e1e0: 0x00032480 0x4004b05c 0x0637481c 0x47138000 0x8ff9fff6 0x27b7c81c 0x87936000 0x4b940007
> 4087e200: 0x43000000 0x4202c4ff 0x420195af 0x4086f80e 0x647d9b54 0x00020000 0x00010020 0x00020020
> 4087e220: 0x42020020 0x42000020 0x0000c4e0 0x00019590 0x00010000 0x00140000 0x00000000 0x40880000
> 4087e240: 0x000038e0 0x4004b05c 0x0000ffff 0xffffffff 0x00000000 0x4087e3a0 0x00000000 0x4086faba
> 4087e260: 0x2def1b7c 0x188e5179 0x00010000 0x00140000 0x00010000 0x200205e9 0x4080217a 0x000000ee
> 4087e280: 0x0000000d 0x00ffff00 0x01000000 0x42020020 0x0000c4e0 0x40800000 0x00003b10 0x42000020
> 4087e2a0: 0x00019590 0x40803b10 0x00007dbc 0x4080b8d0 0x00001100 0x00000000 0x00000000 0x00000000
> 4087e2c0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
> 4087e2e0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
> 4087e300: 0x00000000 0x00000000 0x00000000 0x00010020 0x0001c508 0x00020020 0x000395b8 0x0004137c
> 4087e320: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
> 4087e340: 0x00000000 0x00000000 0x00000000 0x000324a0 0xe724310e 0xc721bc64 0xdaf8fc49 0x63d44ead
> 4087e360: 0x647d9b54 0x600a5267 0xc43ced59 0x5706888f 0xffffffff 0x00000010 0x0000000a 0x40880000
> 4087e380: 0x00000000 0x000000cc 0x000038f0 0x4086c476 0xef906181 0xe760d1f1 0x00000000 0x00000000
> 4087e3a0: 0x0000e000 0x00002000 0x00000000 0x00000000 0x00000000 0x00000000 0x00010000 0x00140000
> 4087e3c0: 0x00150000 0x00140000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
> 4087e3e0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
> 4087e400: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
> 4087e420: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000002 0x00000000
> 4087e440: 0x60002000 0x00000100 0x00000020 0x4001b1c0 0xc5b8d782 0x2163fb0a 0xdb9b1d92 0x00000000
> 4087e460: 0x60645907 0x00000000 0x00010000 0x200203e9 0x4086c410 0x40875728 0x00000594 0x000000ee
> 4087e480: 0x0000000d 0x00ffff00 0x01000000 0x420038f0 0x7533885e 0x8c5a9bca 0x8a57e948 0x675777fb
> 4087e4a0: 0xefe1f630 0x37537b19 0x8f3a9fa2 0x0a23ea69 0x1abef30c 0x00000000 0x00000000 0x00000000
> 4087e4c0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x4087e5f0 0x00000101 0x00000002
> 4087e4e0: 0x8c5a9bca 0x8a57e948 0x675777fb 0xefe1f630 0x37537b19 0x8f3a9fa2 0x0a23ea69 0x1abef30c
> 
> 
> 
> ELF file SHA256: 04c3aa2217fcd4f8

BlinkRGB example from the ESP32 package. Don't work!

> E (43120) rmt: rmt_new_tx_channel(210): invalid interrupt priority:-1515870811

Beacon Scanner works!

Discover Connect example don't work!

> Arduino: 1.8.19 (Windows 10), Board: "ESP32C6 Dev Module, Disabled, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 160MHz (WiFi), QIO, 80MHz, 4MB (32Mb), 921600, None, Disabled"
> 
> DiscoverConnect:26:4: error: #error Serial Bluetooth not available or not enabled. It is only available for the ESP32 chip.
> 
>    26 |   #error Serial Bluetooth not available or not enabled. It is only available for the ESP32 chip.
> 
>       |    ^~~~~
> 
> exit status 1
> 
> #error Serial Bluetooth not available or not enabled. It is only available for the ESP32 chip.


Any idea what is wrong?

**Comment by lbernstone** [comment]
> Discover Connect example don't work!
> 
> > Arduino: 1.8.19 (Windows 10), Board: "ESP32C6 Dev Module, Disabled, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 160MHz (WiFi), QIO, 80MHz, 4MB (32Mb), 921600, None, Disabled"
> > DiscoverConnect:26:4: error: #error Serial Bluetooth not available or not enabled. It is only available for the ESP32 chip.
> > 26 |   #error Serial Bluetooth not available or not enabled. It is only available for the ESP32 chip.
> > ```
> >   |    ^~~~~
> > ```
> > 
> > 
> > exit status 1
> > #error Serial Bluetooth not available or not enabled. It is only available for the ESP32 chip.
> 
> Any idea what is wrong?

Bluetooth classic (as used in BT Serial) is only available on esp32, exactly as the message says.

**Comment by ITstreet1** [comment]
Fine.
What about neopixel?

**Comment by lbernstone** [comment]
RMT has been refactored.  Try https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP32/examples/RMT/RMTWriteNeoPixel/RMTWriteNeoPixel.ino

**Comment by ITstreet1** [comment]
This example blinks the Neopixel in some sort of blue (turquoise, or so).

Bottom line, will the example be included in the package?
Why the Adafruit_Neopixel examples does not work? Will it be also fixed?

**Comment by lbernstone** [comment]
Adafruit_Neopixel is directly using IDF.  They will need to update their library to work with IDF 5.1.
The examples located in https://github.com/espressif/arduino-esp32/tree/master/libraries/ESP32/examples/RMT are what will be included.

**Comment by TD-er** [comment]
> [...] Why the Adafruit_Neopixel examples does not work? Will it be also fixed?

Right now I am looking into what is needed for the Adafruit Neopixel library to support the new RMT interface.
Not sure how much effort this needs as I have never looked into the code of this library myself.

**Comment by me-no-dev** [comment]
RMT issue is a bug. Will be fixed in next release. Till then you can add on like `540` in `esp32-hal-rmt.c` the following code: `rx_cfg.intr_priority = 0;`

**Comment by ITstreet1** [comment]
This remains:

> When connected over the USB port it shows up as an ESP32H2 Dev Module.

When can be expected the main functionalities of C6 such as Zigbee, Thread, etc, to be available?

I have an H2, so I will try this one, too. 

> > [...] Why the Adafruit_Neopixel examples does not work? Will it be also fixed?
> 
> Right now I am looking into what is needed for the Adafruit Neopixel library to support the new RMT interface. Not sure how much effort this needs as I have never looked into the code of this library myself.

I have updated lib, but it is not working

**Comment by TD-er** [comment]
> RMT issue is a bug. Will be fixed in next release. Till then you can add on like `540` in `esp32-hal-rmt.c` the following code: `rx_cfg.intr_priority = 0;`

Could this also be the reason the C3 will boot-loop whenever I have any Adafruit NeoPixel library even included in a build with ESP-IDF 5.x? Even though this library clearly only should be using TX-mode...
Or do you mean adding `tx_cfg.intr_priority = 0;` around line 516?

**Comment by Jason2866** [comment]
Imho talking about 3rd party non working libs is off topic here. The right place to address is the github of the not working lib.

**Comment by ITstreet1** [comment]
> RMT issue is a bug. Will be fixed in next release. Till then you can add on like `540` in `esp32-hal-rmt.c` the following code: `rx_cfg.intr_priority = 0;`

With adding this line on 540 I got the same as before. By adding it on a line 516, I get this:

> Arduino: 1.8.19 (Windows 10), Board: "ESP32C6 Dev Module, Disabled, Enabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 160MHz (WiFi), QIO, 80MHz, 4MB (32Mb), 921600, None, Disabled"
> 
> C:\Users\dekip\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-alpha2\cores\esp32\esp32-hal-rmt.c: In function 'rmtInit':
> 
> C:\Users\dekip\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-alpha2\cores\esp32\esp32-hal-rmt.c:516:9: error: 'rx_cfg' undeclared (first use in this function); did you mean 'tx_cfg'?
> 
>   516 |         rx_cfg.intr_priority = 0;
> 
>       |         ^~~~~~
> 
>       |         tx_cfg
> 
> C:\Users\dekip\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-alpha2\cores\esp32\esp32-hal-rmt.c:516:9: note: each undeclared identifier is reported only once for each function it appears in
> 
> exit status 1
> 
> Error compiling for board ESP32C6 Dev Module.
> 
> 
> 
> This report would have more information with
> "Show verbose output during compilation"
> option enabled in File -> Preferences.

@Jason2866 
If the reason the lib doesn't work is bugs in the C6 support package that should be fixed as @TD-er mentioned, then this is not off-topic.
https://github.com/espressif/arduino-esp32/issues/7713#issuecomment-1774221708

**Comment by me-no-dev** [comment]
@ITstreet1 this: https://github.com/espressif/arduino-esp32/commit/730d827553cf3ae357eb62a3b66f97b13a255177

**Comment by ITstreet1** [comment]
Yes. On this exact place, I still get this text/number wall from above.

**Comment by Jason2866** [comment]
It is an Adafruit Neopixel Lib issue. RMT is working well with an Arduino 3.0.0 adopted version of Makuna Neopixelbus lib. Tested and running with ESP32, ESP32-S2, ESP32-S3, ESP32-C3 and ESP32-C6.

**Comment by ITstreet1** [comment]
https://github.com/espressif/arduino-esp32/issues/7713#issuecomment-1774174954
This works too, I believe.
My point is about:
https://github.com/espressif/arduino-esp32/issues/7713#issuecomment-1774221708

I believe you suggest me to check the Adafruit GitHub repo? Or to wait for you to fix the bug, if there is a bug after all?

**Comment by Jason2866** [comment]
Yes, the example from espressif is working too. I won't fix Adafruit Neopixel, since i don't use.
So the way is open an issue in Github of Adafruit Neopixel.

**Comment by Jason2866** [comment]
This is as it should be
> When connected over the UART port it shows up as a COM port

Can't be fixed either the H2 or the C6 is shown wrong.
> When connected over the USB port it shows up as an ESP32H2 Dev Module.
Obviously, it is not an H2.

The have the same `PID` `VID` no indication to check what device is really connected.

**Comment by VojtechBartoska** [comment]
Neopixel library is not supporting our core (latest IDF yet), here is result from out Library testing. https://github.com/espressif/arduino-esp32/blob/gh-pages/LIBRARIES_TEST.md

**Comment by VojtechBartoska** [comment]
> You can fix this though:
> 
> > When connected over the UART port it shows up as a COM port
> > When connected over the USB port it shows up as an ESP32H2 Dev Module.
> > Obviously, it is not an H2.
> 
> Besides, what are the expectations about Zigbee, Thread, etc? If there are any. :)

Hello, we did initial investigation for supporting those protocols but did not find a proper implementation way. It is not going to be part of 3.0.0, if applicable, we will add it later.

**Comment by MingyaoLiu** [comment]
I can build and upload to ESP32-C6 WROOM on thhe 3.0.0-alpha2. However I can't see the serial monitor. If I change the board from "ESP32C6 Dev Module" to "Spark Fun ESP32C6 wiic Pocket", I can build and upload, and see Serial print out. I also tried building a simple serial print in ESP-IDF in platformio, it also works.

Anyone can point me to how to fix this serial issue? Is there something I can change in board defination to fix this?

Thx.

**Comment by MingyaoLiu** [comment]
> I can build and upload to ESP32-C6 WROOM on thhe 3.0.0-alpha2. However I can't see the serial monitor. If I change the board from "ESP32C6 Dev Module" to "Spark Fun ESP32C6 wiic Pocket", I can build and upload, and see Serial print out. I also tried building a simple serial print in ESP-IDF in platformio, it also works.
> 
> Anyone can point me to how to fix this serial issue? Is there something I can change in board defination to fix this?
> 
> Thx.

ok I fixed it, forgot to enable CDC.

**Comment by rdahlin** [comment]
I have tested the alpha for Arduino IDE and so far it has worked fine.
I have tried to get neopixel to work but understand that it's not possible yet and now I don't get any output in the console when connected to the usb port (not UART) if I don't nave CDC enabled.  When I do so the log output works again but when I do a BLE "characteristic->notify()"  I get "-> E (139991) rmt: rmt_new_tx_channel(211): invalid interrupt priority:1107301138" in the log. I have not got this before. I have printed out debug values of that I just set for the characteristic and before that and nothing else has been printed.

If i disable CDC i get "-> controller lib commit: [3ff2145]" instead.

Am I doning something wrong here or is it some bugs that causes this?

**Comment by me-no-dev** [comment]
@rdahlin this has already been fixed in master, Will be part of next release.

**Comment by mightChamp** [comment]
Has anyone tested SPI communication in ESP32-C6 in Alpha2 release ?
SPI fails at spiStartBus(), as spi_num >= SPI_COUNT
here SPI_COUNT is set to 1 for ESP32-C6, and default HSPI value is 1, So SPI always fails to init.

**Comment by rdahlin** [comment]
> @rdahlin this has already been fixed in master, Will be part of next release.

@me-no-dev : ETA?

**Comment by me-no-dev** [comment]
soon. maybe week or so.

**Comment by mightChamp** [comment]
> @mightChamp ESP32C6 have only one general purpose SPI and it is FSPI. Thanks for pointing out that the define for HSPI exists there for C6.

![image](https://github.com/espressif/arduino-esp32/assets/45094530/3b8cebad-b7e2-4bbe-afd8-e824aa162b5f)

It shows 3 SPI port in ESP32-C66.

Is FSPI used by FLASH in ESP32-C6?

**Comment by mightChamp** [comment]
> @mightChamp ESP32C6 have only one general purpose SPI and it is FSPI. Thanks for pointing out that the define for HSPI exists there for C6.

Is FSPI used by FLASH in ESP32-C6?

**Comment by elgerg** [comment]
> > You can fix this though:
> > > When connected over the UART port it shows up as a COM port
> > > When connected over the USB port it shows up as an ESP32H2 Dev Module.
> > > Obviously, it is not an H2.
> > 
> > 
> > Besides, what are the expectations about Zigbee, Thread, etc? If there are any. :)
> 
> Hello, we did initial investigation for supporting those protocols but did not find a proper implementation way. It is not going to be part of 3.0.0, if applicable, we will add it later.

Oh thats a real shame. I would really like to use the H2 and C6 with Zigbee

**Comment by P-R-O-C-H-Y** [comment]
@elgerg I am currently working on a solution to add a proper support for Zigbee protocol. You can follow this open ticket #8807, which is being updated.

**Comment by elgerg** [comment]
> @elgerg I am currently working on a solution to add a proper support for Zigbee protocol. You can follow this open ticket #8807, which is being updated.

That's great news!
Thanks

**Comment by dkull** [comment]
https://blog.espressif.com/announcing-the-arduino-esp32-core-version-3-0-0-3bf9f24e20d4
Beautiful!

**Comment by ITstreet1** [comment]
Ok,
I tried a simple I2C scanner to scan a connected Oled screen. No I2C devices were found!

According to the diagram, the SDA and SCL pins are 6 and 7 of the pinout.
https://docs.espressif.com/projects/espressif-esp-dev-kits/en/latest/_images/esp32-c6-devkitm-1-pin-layout.png

Is this an issue, or I made a mistake? From what I see on the table with the support overview, there is nothing for I2C.

**Comment by P-R-O-C-H-Y** [comment]
Hi @ITstreet1, in the `pins_arduino.h` file for ESP32-C6 the default pin for I2C are:
```
static const uint8_t SDA = 23;
static const uint8_t SCL = 22;
```
Can you try it please using those pins? :)

**Comment by ITstreet1** [comment]
Tried both. Nothing.
Tried the BME280. Nothing
I can upload picture if you wish.

**Comment by MingyaoLiu** [comment]
i2C works fine on my esp32c6, tested scd41 and bme680

**Comment by toddbu** [comment]
Hey folks, sorry to ask such a n00bie question, but here I go anyway...

I've recently receive five C6 development boards and am eager to try them out. Looking at the announcement link above (https://blog.espressif.com/announcing-the-arduino-esp32-core-version-3-0-0-3bf9f24e20d4), I'm told to add `https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json` to the boards manager. So I did this. Just for good measure, I also removed `https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json` so I should only be retrieving one ESP board definition. After watching the new board file download, I went into the boards manager and didn't see anything with "c6" in the board name. If I go to https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json and search the file, I find nothing when searching for "-c6", even though "-c3" returns 17 hits. Am I doing something wrong here? Do I have the wrong URL?

**Comment by ITstreet1** [comment]
You don't need this URL. All you have to do is install the ESP32 through the boards manager.

**Comment by ITstreet1** [comment]
@P-R-O-C-H-Y 
Tried a few devices, tried both suggested pins, tried a few wires, and tried magic, but nothing. There is nothing on the I2C scanner.
Any suggestions? Should I try some other pins?

**Comment by toddbu** [comment]
Thanks @ITstreet1. That worked perfectly

**Comment by atchoo78** [comment]
Hi! I'm getting error messages to the moon and back trying to make BLE MIDI (NimBLE) work on a ESP32-C6 devkit board. Any insights on this, anyone?

On the bright side, the updated example sketches for ESP32 BLE seems to compile and upload to this board now 😘

**Comment by Jason2866** [comment]
@atchoo78 try [h2zero NimBLE](https://github.com/h2zero/esp-nimble-cpp) it works well in for us in Tasmota and the C6.

**Comment by atchoo78** [comment]
> @atchoo78 try [h2zero NimBLE](https://github.com/h2zero/esp-nimble-cpp) it works well for us in Tasmota and the C6.

Already using it (unfortunately), but thanks. The MIDI via BLE rabbit hole is just way too complicated for me to debug or even grasp right now, but maybe someone else can. Here's the last error message I got before I kinda gave up the whole thing. 

```
(...)/Arduino/libraries/BLE-MIDI/src/hardware/BLEMIDI_ESP32.h: In member function 'virtual void bleMidi::MyCharacteristicCallbacks::onWrite(BLECharacteristic*)':
(...)/Arduino/libraries/BLE-MIDI/src/hardware/BLEMIDI_ESP32.h:116:55: error: conversion from 'String' to non-scalar type 'std::string' {aka 'std::__cxx11::basic_string<char>'} requested
	116 |	std::string rxValue = characteristic->getValue();
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~^~

exit status 1

```

Keeping my fingers crossed for BLE MIDI support on the ESP32-C6 soon 😅🙏🏼🧞‍♂️
Thanks

**Comment by lbernstone** [comment]
@atchoo78 This is code that compiles in 3.0.0 for other variants?

**Comment by atchoo78** [comment]
> @atchoo78 This is code that compiles in 3.0.0 for other variants?

@lbernstone Correct. Compiling and running on my Esp32 v1 devkit and ESP32-C3.

**Comment by Jason2866** [comment]
@atchoo78 Using `esp-nimble-cpp` it is the IDF variant but can be used with Arduino too? The Arduino version from h2zero does not support the C6/C2.

**Comment by NorthLight-EWR** [comment]
Hi all,
unfortunately, my first idea to use arduino needs zigbee, so I would have to use this development branch.

What do I have to take care for buying it?

So if I buy a board now, p.e. like [this](https://www.ebay.de/itm/175982398648?mkcid=16&mkevt=1&mkrid=707-127634-2357-0&ssspo=HYWYthbaRNe&sssrc=2047675&ssuid=rWJ4m5W_Q7G&widget_ver=artemis&media=COPY), how do I know, i can use it with Arduino-esp v3.x.x ?

Finally, i need to connect an i2c device..

**Comment by Jason2866** [comment]
@ITstreet1  Does not work, doesnt help. Please provide the code you have tried.

**Comment by Jason2866** [comment]
Example code!!!

**Comment by Jason2866** [comment]
@ITstreet1 Used the i2c scanner from examples and connected a BME280 to the C6 GPIOs 22 / 23 and get this
```
Scanning for I2C devices ...
I2C device found at address 0x76
```

**Comment by ITstreet1** [comment]
> And just to add, I have an ESP32 C6 DevKitM 1 V1.0 dev kit
> I have ESP32 H2, too. Nothing on I2C.

I have Espressif's test samples, both C6 and H2. Could there be issues with those? If you have any suggestions for me to try, to debug this, I am all ears.

**Comment by Jason2866** [comment]
Mine is an espressif ESP32 C6 DevKitM-1 too

**Comment by toddbu** [comment]
Is anyone aware that `gpio_wakeup_enable` is missing from `3.0.0-alpha3`? The code snippet below compiles on an `ESP32C3 Dev Module` board under `2.0.11` but breaks on 3.0.0-alpha3` with the message `Compilation error: 'gpio_wakeup_enable' was not declared in this scope`. The function should still be supported according to the API docs, even for the C6 (https://docs.espressif.com/projects/esp-idf/en/latest/esp32c6/api-reference/system/sleep_modes.html).

```
#define UPPER_DOOR 0

void setup() {
  gpio_wakeup_enable((gpio_num_t) UPPER_DOOR, GPIO_INTR_HIGH_LEVEL);
}

void loop() {
}
```

**Comment by Jason2866** [comment]
Dont see the call you try in the API list https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32c6/api-reference/system/sleep_modes.html
There is `esp_sleep_enable_gpio_wakeup`

**Comment by toddbu** [comment]
Those are different calls. You use `gpio_wakeup_enable` to set the pin(s) that you want to be able to wakeup on, then call `esp_sleep_enable_gpio_wakeup` to turn on the wakeup for those pins. It's documented at https://docs.espressif.com/projects/esp-idf/en/latest/esp32c6/api-reference/peripherals/gpio.html. I've had code successfully using these two API calls in succession running on an esp32-c3 for several months, so the API call has broken in `3.0.0-alpha3` from `2.0.11`

**Comment by maxdd** [comment]
Hello,
In the arduino esp32 sdk readme about the zigbee example i've read that C6 is dramatically worse in terms of performance since it uses a single RF path (shared between zigbee and wifi) wrt to whatever available combo. Are you aware of which performance penality are we talking about?

**Comment by Jason2866** [comment]
@maxdd There is probably no easy answer possible. It depends how fast is the switching between wifi and zigbee. If this is implemented well, there is a chance wifi and zigbee performs really well with no really performance penalty. If this happens not smooth performance will be horrible.

**Comment by simogaspa84** [comment]
Hi guys news about C6 support in platformio ? How is going the integration process? Thanks

**Comment by TD-er** [comment]
When using the platform_packages made by @Jason2866 you can use the C6 perfectly fine.
I've been working on the C2/C6 for a few months now and all features already present in the other modules, including wifi6 (802.11ax) are working just fine.
I haven't tested the 802.15.4 yet, so can't speak about it from experience.

**Comment by TD-er** [comment]
see: https://github.com/letscontrolit/ESPEasy/blob/mega/platformio_core_defs.ini

and newer builds made by Jason:
https://github.com/Jason2866/esp32-arduino-lib-builder/releases
You need the release/v5.1 ones as `platform_package`
and pay attention to use the ones without "solo1" in the name

**Comment by Jason2866** [comment]
Build from yesterday. Complete setup just use this
```
 platform = https://github.com/tasmota/platform-espressif32/releases/download/2024.01.11/platform-espressif32.zip
```

**Comment by gtom84** [comment]
I can't seems to get to work ESP32-C6 with Arduino IDE 2.3.2 using Alpha3 Board Library from Dev channel `https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json`

I installed alpha3 to Arduino IDE, selected ESP32C6 board. I created a super simple sketch, no external libraries. Compilation and upload are OK but there is nothing coming out of ESP32 on Serial port.

```
void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:
  Serial.println("Hello world");
  delay(1000);
}

```

If I compile this example (in C) or an example from ESP-IDF, compile it with `idf.py build` and upload it to ESP32-C6 then it works OK. It must be something with Arduino IDE. Is there another step required after installation and selected of the board in Arduino IDE?

**Comment by TD-er** [comment]
What is your "Serial port"?
Do you use an USB to serial chip or is the USB connector connected directly to the ESP32-C6?
If you are using this "native USB" serial port you need to enable to use the  HWCDC port.
Maybe there are also board definitions which have this already set, so you then only need to select a different board.

**Comment by JannikRoesch** [comment]
> I can't seems to get to work ESP32-C6 with Arduino IDE 2.3.2 using Alpha3 Board Library from Dev channel `https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json`
> 
> 
> 
> I installed alpha3 to Arduino IDE, selected ESP32C6 board. I created a super simple sketch, no external libraries. Compilation and upload are OK but there is nothing coming out of ESP32 on Serial port.
> 
> 
> 
> ```
> 
> void setup() {
> 
>   // put your setup code here, to run once:
> 
> 
> 
> }
> 
> 
> 
> void loop() {
> 
>   // put your main code here, to run repeatedly:
> 
>   Serial.println("Hello world");
> 
>   delay(1000);
> 
> }
> 
> 
> 
> ```
> 
> 
> 
> If I compile this example (in C) or an example from ESP-IDF, compile it with `idf.py build` and upload it to ESP32-C6 then it works OK. It must be something with Arduino IDE. Is there another step required after installation and selected of the board in Arduino IDE?
> 
> 
> 
> 

I think you forgot to call something like "Serial.begin(115200)" in your setup function.

**Comment by anyonecan** [comment]
> What is your "Serial port"? Do you use an USB to serial chip or is the USB connector connected directly to the ESP32-C6? If you are using this "native USB" serial port you need to enable to use the HWCDC port. Maybe there are also board definitions which have this already set, so you then only need to select a different board.

I have a problem similar to gtom84. I have the ESP32-C6-DevKitM-1 board and I've added Serial.print s to the Blink example, with boards selected from the 3.0.0.a library. I'm using the USB connector and uploads are OK so I think the basic connection is good. Depending on which board is selected, I can get the LED or Serial to work, or neither works, but not all the same time.

It should be simple to fix the LED problem, but I haven't succeeded yet. Would say the native port is already enabled? If not, how do I do that?

**Comment by Jason2866** [comment]
@slaesh Only can say Tasmota works stable with the C6 (test run a few days) no crash or reboot. Tasmota uses mqtt by default. There was a BME280 sensor connected, providing sensor data every 5 minutes via mqtt.

**Comment by slaesh** [comment]
thanks @Jason2866 @TD-er , works now fine. had a weird mis-configuration somehow. was going to a newer platform_packages build and got weird compile time errors like saying that the WiFiEvent_t type is missing and other stuff.. after completly removing and re-installing platformio and all deps it went away.. Oo dunno what this was, but now it works as expected! :)) thanks for your feedback, so I could narrow it down faster :)

**Comment by probonopd** [comment]
Cool. Is there zigbee support and an example?

**Comment by VojtechBartoska** [comment]
@probonopd ported ESP-IDF examples to Arduino are here: https://github.com/espressif/arduino-esp32/tree/master/libraries/ESP32/examples/Zigbee

**Comment by ITstreet1** [comment]
Top!!!
Will it be available through a package update soon?

**Comment by kroon040** [comment]
Hi,

I also run tasmota with and esp32-c6. It connects to g-HE (AX), but I noticed ping from a client to the esp32c6 I got a ttl of 63 instead of 254? @Jason2866 is this also with your device, and is it tasmota of arduino related?

**Comment by TD-er** [comment]
What do you mean by a TTL of 63?

Not sure if this also applies when connecting via AX (thus HE instead of HT), but I also have set this in my (ESPEasy) code:
```c++
  // HT20 = 20 MHz channel width.
  // HT40 = 40 MHz channel width.
  // In theory, HT40 can offer upto 150 Mbps connection speed.
  // However since HT40 is using nearly all channels on 2.4 GHz WiFi,
  // Thus you are more likely to experience disturbances.
  // The response speed and stability is better at HT20 for ESP units.
  esp_wifi_set_bandwidth(WIFI_IF_STA, WIFI_BW_HT20);
```
This bandwidth enum does only have HT20/HT40.

I noticed that ESP32-xx does perform less 'snappy' when using HT40.
So maybe this also does help something when you may experience poor response when connected via AX?

I did add WiFi 6 support after setting this bandwidth limitation so I never tried without.

**Comment by Jason2866** [comment]
@kroon040 i have no wifi 6 equipment

**Comment by kroon040** [comment]
> What do you mean by a TTL of 63?
> 
Here exmapl of 3 devices , the last with esp32-c6, all with tasmota . 
Fist is a normal esp32 device

ping 192.168.170.68
PING 192.168.170.68 (192.168.170.68) 56(84) bytes of data.
64 bytes from 192.168.170.68: icmp_seq=1 ttl=254 time=33.1 ms
64 bytes from 192.168.170.68: icmp_seq=2 ttl=254 time=7.44 ms
64 bytes from 192.168.170.68: icmp_seq=3 ttl=254 time=75.6 ms
64 bytes from 192.168.170.68: icmp_seq=4 ttl=254 time=8.09 ms

An esp32-c3 device

PING 192.168.170.35 (192.168.170.35) 56(84) bytes of data.
64 bytes from 192.168.170.35: icmp_seq=1 ttl=254 time=3.00 ms
64 bytes from 192.168.170.35: icmp_seq=2 ttl=254 time=2.81 ms
64 bytes from 192.168.170.35: icmp_seq=3 ttl=254 time=2.75 ms


The esp32-c6 device

ping 192.168.170.103
PING 192.168.170.103 (192.168.170.103) 56(84) bytes of data.
64 bytes from 192.168.170.103: icmp_seq=1 ttl=63 time=36.6 ms
64 bytes from 192.168.170.103: icmp_seq=2 ttl=63 time=60.2 ms
64 bytes from 192.168.170.103: icmp_seq=3 ttl=63 time=80.5 ms

**Comment by Jason2866** [comment]
Looks like a different setting choosen for TTL with the C6. Will have no impact in real life.
The high ping times in relation to the other devices is something which is not good.

**Comment by TD-er** [comment]
I hardly ever pay attention to the TTL of ping.
I only use it for traceroute/tracepath kinda stuff as that's using the TTL of a ping to get some idea of the route a packet takes to reach some host. 
On each 'hop' the TTL is decreased with 1 (typically) and as soon as the TTL reaches 0, it is discarded.
I have no clue why it is set to 63 here as I doubt it has already taken 192 hops to reach your ESP and I assume you did do these pings from the same machine?
If not, then that's probably your answer as the initial TTL value can be different from platform to platform.

I doubt any hop on the route to your ESP will 'restart' the TTL at some much lower value like 128 or 64 as this can mess with tools like traceroute/tracepath. (though I doubt you will ever need more than ~30 hops to reach anything)

**Comment by 229311JK** [comment]
> Top!!! Will it be available through a package update soon?

    To the Devs, you guys do great work & thanks 4 ESP-32 H2/C6 support!  Any simple way to contribute, please contact me.  Very talented, just want to crawl before sprinting...  To all you others, here you go:  https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json.  Cut/paste to your IDE board manager URLS, download and go to your board options.  Credits to Sparkfun Industries, https://docs.sparkfun.com...

**Comment by 229311JK** [comment]
> Top!!! Will it be available through a package update soon?

Look below....

**Comment by 229311JK** [comment]
To the Devs, you guys do great work & thanks 4 ESP-32 H2/C6 support!  
Any simple way to contribute, please contact me.  Very talented, just want to crawl before sprinting...  

     To all you others, here you go:  https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json.  Cut/paste to your IDE board manager URLS, download and go to your board options.  Credits to Sparkfun Industries, https://docs.sparkfun.com...

**Comment by 229311JK** [comment]
To the Devs, you guys do great work & thanks 4 ESP-32 H2/C6 support!  Any simple way to contribute, please contact me.  Very talented, just want to crawl before sprinting...  To all you others, here you go:  https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json.  Cut/paste to your IDE board manager URLS, download and go to your board options.  Credits to Sparkfun Industries, https://docs.sparkfun.com...

**Comment by VojtechBartoska** [comment]
Hello all,

version 3.0.0 stable was released yesterday and it includes support for ESP32-C6.

I am closing this ticket as solved and if you find any new issue related to ESP32-C6 chip, please open a new ticket.

Thanks for your contribution.
